<!DOCTYPE html>
<html lang="zh-Hans">
<head>

    <!--[if lt IE 9]>
        <style>body {display: none; background: none !important} </style>
        <meta http-equiv="Refresh" Content="0; url=//outdatedbrowser.com/" />
    <![endif]-->

<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
<meta name="format-detection" content="telephone=no" />
<meta name="author" content="紫色的紫" />



<meta property="og:type" content="article">
<meta property="og:title" content="GATK_bamfile">
<meta property="og:url" content="http://yoursite.com/2016/11/29/GATK-bamfile/index.html">
<meta property="og:site_name" content="Purple Violet">
<meta property="og:image" content="http://yoursite.com/images/BP_somatic_workflow_mutect1.png">
<meta property="og:image" content="http://yoursite.com/images/processor.png">
<meta property="og:image" content="http://yoursite.com/images/processor2.png">
<meta property="og:image" content="http://yoursite.com/images/processor3.png">
<meta property="og:image" content="http://yoursite.com/image/mutectOutput.png">
<meta property="og:updated_time" content="2016-12-18T10:51:18.271Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="GATK_bamfile">
<meta name="twitter:image" content="http://yoursite.com/images/BP_somatic_workflow_mutect1.png">

<link rel="apple-touch-icon" href= "/apple-touch-icon.png">


    <link rel="alternate" href="/atom.xml" title="Purple Violet" type="application/atom+xml">



    <link rel="shortcut icon" href="/favicon.png">



    <link href="//cdn.bootcss.com/animate.css/3.5.1/animate.min.css" rel="stylesheet">



    <link href="//cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.min.css" rel="stylesheet">



    <script src="//cdn.bootcss.com/pace/1.0.2/pace.min.js"></script>
    <link href="//cdn.bootcss.com/pace/1.0.2/themes/blue/pace-theme-minimal.css" rel="stylesheet">


<link rel="stylesheet" href="/css/style.css">



<link href="//cdn.bootcss.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet">


<title>GATK_bamfile | Purple Violet</title>

<script src="//cdn.bootcss.com/jquery/2.2.4/jquery.min.js"></script>
<script src="//cdn.bootcss.com/clipboard.js/1.5.10/clipboard.min.js"></script>

<script>
    var yiliaConfig = {
        fancybox: true,
        animate: true,
        isHome: false,
        isPost: true,
        isArchive: false,
        isTag: false,
        isCategory: false,
        fancybox_js: "//cdn.bootcss.com/fancybox/2.1.5/jquery.fancybox.min.js",
        scrollreveal: "//cdn.bootcss.com/scrollReveal.js/3.1.4/scrollreveal.min.js",
        search: undefined
    }
</script>


    <script> yiliaConfig.jquery_ui = [false]; </script>



    <script> yiliaConfig.rootUrl = "\/";</script>






</head>
<body>
  <div id="container">
    <div class="left-col">
    <div class="overlay"></div>
<div class="intrude-less">
    <header id="header" class="inner">
        <a href="/" class="profilepic">
            <img src="/img/avatar.png" class="animated zoomIn">
        </a>
        <hgroup>
          <h1 class="header-author"><a href="/">紫色的紫</a></h1>
        </hgroup>

        

        


        
            <div id="switch-btn" class="switch-btn">
                <div class="icon">
                    <div class="icon-ctn">
                        <div class="icon-wrap icon-house" data-idx="0">
                            <div class="birdhouse"></div>
                            <div class="birdhouse_holes"></div>
                        </div>
                        <div class="icon-wrap icon-ribbon hide" data-idx="1">
                            <div class="ribbon"></div>
                        </div>
                        
                        <div class="icon-wrap icon-link hide" data-idx="2">
                            <div class="loopback_l"></div>
                            <div class="loopback_r"></div>
                        </div>
                        
                        
                        <div class="icon-wrap icon-me hide" data-idx="3">
                            <div class="user"></div>
                            <div class="shoulder"></div>
                        </div>
                        
                    </div>
                    
                </div>
                <div class="tips-box hide">
                    <div class="tips-arrow"></div>
                    <ul class="tips-inner">
                        <li>菜单</li>
                        <li>标签</li>
                        
                        <li>友情链接</li>
                        
                        
                        <li>关于我</li>
                        
                    </ul>
                </div>
            </div>
        

        <div id="switch-area" class="switch-area">
            <div class="switch-wrap">
                <section class="switch-part switch-part1">
                    <nav class="header-menu">
                        <ul>
                        
                            <li><a href="/">主页</a></li>
                        
                            <li><a href="/archives/">所有文章</a></li>
                        
                            <li><a href="/tags/">标签云</a></li>
                        
                            <li><a href="/about/">关于我</a></li>
                        
                        </ul>
                    </nav>
                    <nav class="header-nav">
                        <ul class="social">
                            
                                <a class="fa Email" href="mailto:1479205243@qq.com" title="Email"></a>
                            
                                <a class="fa GitHub" href="#" title="GitHub"></a>
                            
                                <a class="fa RSS" href="/atom.xml" title="RSS"></a>
                            
                        </ul>
                    </nav>
                </section>
                
                
                <section class="switch-part switch-part2">
                    <div class="widget tagcloud" id="js-tagcloud">
                        
                    </div>
                </section>
                
                
                
                <section class="switch-part switch-part3">
                    <div id="js-friends">
                    
                      <a class="main-nav-link switch-friends-link" href="https://hexo.io">Hexo</a>
                    
                      <a class="main-nav-link switch-friends-link" href="https://pages.github.com/">GitHub</a>
                    
                      <a class="main-nav-link switch-friends-link" href="http://moxfive.xyz/">MOxFIVE</a>
                    
                    </div>
                </section>
                

                
                
                <section class="switch-part switch-part4">
                
                    <div id="js-aboutme">2016 北京航空航天大学自动化科学与电气工程学院</div>
                </section>
                
            </div>
        </div>
    </header>                
</div>
    </div>
    <div class="mid-col">
      <nav id="mobile-nav">
      <div class="overlay">
          <div class="slider-trigger"></div>
          <h1 class="header-author js-mobile-header hide"><a href="/" title="回到主页">紫色的紫</a></h1>
      </div>
    <div class="intrude-less">
        <header id="header" class="inner">
            <a href="/" class="profilepic">
                <img src="/img/avatar.png" class="animated zoomIn">
            </a>
            <hgroup>
              <h1 class="header-author"><a href="/" title="回到主页">紫色的紫</a></h1>
            </hgroup>
            
            <nav class="header-menu">
                <ul>
                
                    <li><a href="/">主页</a></li>
                
                    <li><a href="/archives/">所有文章</a></li>
                
                    <li><a href="/tags/">标签云</a></li>
                
                    <li><a href="/about/">关于我</a></li>
                
                <div class="clearfix"></div>
                </ul>
            </nav>
            <nav class="header-nav">
                        <ul class="social">
                            
                                <a class="fa Email" target="_blank" href="mailto:1479205243@qq.com" title="Email"></a>
                            
                                <a class="fa GitHub" target="_blank" href="#" title="GitHub"></a>
                            
                                <a class="fa RSS" target="_blank" href="/atom.xml" title="RSS"></a>
                            
                        </ul>
            </nav>
        </header>                
    </div>
    <link class="menu-list" tags="标签" friends="友情链接" about="关于我"/>
</nav>
      <div class="body-wrap"><article id="post-GATK-bamfile" class="article article-type-post" itemscope itemprop="blogPost">
  
    <div class="article-meta">
      <a href="/2016/11/29/GATK-bamfile/" class="article-date">
      <time datetime="2016-11-29T07:52:15.177Z" itemprop="datePublished">2016-11-29</time>
</a>


    </div>
  
  <div class="article-inner">
    
      <input type="hidden" class="isFancy" />
    
    
      <header class="article-header">
        
  
    <h1 class="article-title" itemprop="name">
      GATK_bamfile
    </h1>
  

      </header>
      
      <div class="article-info article-info-post">
        

        
        <div class="clearfix"></div>
      </div>
      
    
    <div class="article-entry" itemprop="articleBody">
      
          
        <p><excerpt in="" index="" |="" 首页摘要=""><br><a id="more"></a></excerpt></p>
<p><the rest="" of="" contents="" |="" 余下全文=""><br><strong>操作对象是两个文件</strong><br>文件1：病人TCGA-2V-A95S癌组织样本的数据<br>文件路径：/data2/TCGA/LIHC/WXS/0a2df548-7c8c-46c5-9290-2e4d85604c17/TCGA-2V-A95S-01A-11D-A36X-10_Illumina.bam<br>文件2：同一病人TCGA-2V-A95S血细胞（正常）样本数据<br>文件路径：/data2/TCGA/LIHC/WXS/5c864e10-843b-41b8-ae10-2a78236b393c/TCGA-2V-A95S-10D-01D-A370-10_Illumina.bam<br>对文件1进行操作<br><strong>command1：</strong><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtools view -H /data2/TCGA/LIHC/WXS/0a2df548-7c8c-46c5-9290-2e4d85604c17/TCGA-2V-A95S-01A-11D-A36X-10_Illumina.bam | grep &apos;@RG&apos;</div></pre></td></tr></table></figure></the></p>
<p><strong>result1:</strong><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">@RG     ID:0    PL:Illumina     PU:140928_SN896_0553_AC4HHCACXX_5_ID04  LB:TLVR.2V-A95S-01A-T_1AMP      DS:TCGA Liver   DT:2014-10-10T03:00:00-0500     SM:TCGA-2V-A95S-01A-11D-A36X-10 CN:BCM</div></pre></td></tr></table></figure></p>
<p><strong>说明：</strong><br><strong>command2:</strong><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">samtool view -H /data2/TCGA/LIHC/WXS/0a2df548-7c8c-46c5-9290-2e4d85604c17/TCGA-2V-A95S-01A-11D-A36X-10_Illumina.bam | grep &apos;@PG&apos;</div></pre></td></tr></table></figure></p>
<p><strong>result2:</strong><br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">@PG     ID:GATK IndelRealigner  VN:2.5-2-gf57256b       CL:knownAlleles=[(RodBinding name=knownAlleles source=/stornext/snfs5/next-gen/Illumina/ipipeV3.6/Databases/1KG_InDels/1000G_phase1.indels.b37.vcf.gz), (RodBinding name=knownAlleles2 source=/stornext/snfs5/next-gen/Illumina/ipipeV3.6/Databases/Mills_InDels/Mills_and_1000G_gold_standard.indels.b37.vcf.gz)] targetIntervals=/space1/tmp/11362609.sug-moab/C4HHCACXX-5-ID04.GATKrealign.intervals LODThresholdForCleaning=5.0 consensusDeterminationModel=USE_READS entropyThreshold=0.15 maxReadsInMemory=150000 maxIsizeForMovement=3000 maxPositionalMoveAllowed=200 maxConsensuses=30 maxReadsForConsensuses=120 maxReadsForRealignment=20000 noOriginalAlignmentTags=false nWayOut=null generate_nWayOut_md5s=false check_early=false noPGTag=false keepPGTags=false indelsFileForDebugging=null statisticsFileForDebugging=null SNPsFileForDebugging=null</div><div class="line">@PG     ID:MarkDuplicates       PN:MarkDuplicates       VN:1.93(1476)   CL:net.sf.picard.sam.MarkDuplicates INPUT=[/space1/tmp/11362608.sug-moab/C4HHCACXX-5-ID04_sorted.bam] OUTPUT=C4HHCACXX-5-ID04_marked.bam METRICS_FILE=C4HHCACXX-5-ID04.PCR_Duplicates_metrics.txt ASSUME_SORTED=true TMP_DIR=[/space1/tmp] VALIDATION_STRINGENCY=LENIENT MAX_RECORDS_IN_RAM=2500000 CREATE_INDEX=true    PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates REMOVE_DUPLICATES=false MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 READ_NAME_REGEX=[a-zA-Z0-9]+:[0-9]:([0-9]+):([0-9]+):([0-9]+).* OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 VERBOSITY=INFO QUIET=false COMPRESSION_LEVEL=5 CREATE_MD5_FILE=false</div><div class="line">@PG     ID:bwa  PN:bwa  VN:0.6.2-r126</div><div class="line">@PG     ID:GATK PrintReads      VN:2.5-2-gf57256b       CL:readGroup=null platform=null number=-1 downsample_coverage=1.0 sample_file=[] sample_name=[] simplify=false no_pg_tag=false</div></pre></td></tr></table></figure></p>
<p><strong>说明：</strong><br>对bam文件进行过四次操作，并且越靠前的操作记录，所对应的时间越接近现在，对照BEST PRACTICE的workflow，bam文件已经经过了BWA、MarkDuplicates、和IndelRealigner的处理，已经来到T/N PRE-PROCESSING 的Indel Realignment阶段（基因重组），根据流程图，下一阶段可认为是用MuTect处理。</p>
<p><img src="/images/BP_somatic_workflow_mutect1.png" alt="BP_somatic_workflow_mutect1"><br><strong>剩下的内容在GATK Presentation 里，我tm居然过了一个月才找到最直接的参考资料，蓝瘦，香菇</strong><br>variant discovery 是通过Mutect来完成的，Mutect1.1.7有单独的jar包，使用的代码格式如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">java –jar mutect-­‐1.1.7.jar\</div><div class="line">–T MuTect\</div><div class="line">–R human.fasta\</div><div class="line">–I:normal normal.bam\</div><div class="line">–I:tumor tumor.bam\</div><div class="line">--dbsnp dbsnp137.vcf\</div><div class="line">­‐cosmic cosmic.vcf\</div><div class="line">[–L exome_targets.intervals\]</div><div class="line">–o sample.call_stats.txt\</div><div class="line">­‐coverage_file sample.coverage.wig.txt</div></pre></td></tr></table></figure></p>
<p>可以看到，除了normal和tumor两个bam文件之外，还有dbsnp文件和cosmic的vcf文件，参考序列fasta文件，输出格式是txt。dbsnp文件和cosmic文件以及fasta文件，GATK提供了下载的ftp路径，<a href="http://gatkforums.broadinstitute.org/gatk/discussion/1213/whats-in-the-resource-bundle-and-how-can-i-get-it" target="_blank" rel="external">相关网页</a><br>由于Mutect1.1.7的输出格式是txt,我们选择可以输出vcf格式的Mutect2版本，Mutect2是在GenomeAnalysisTK.jar包里的，调用格式为<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div></pre></td><td class="code"><pre><div class="line">java</div><div class="line">     -jar GenomeAnalysisTK.jar \</div><div class="line">     -T MuTect2 \</div><div class="line">     -R reference.fasta \</div><div class="line">     -I:tumor tumor.bam \</div><div class="line">     -I:normal normal.bam \</div><div class="line">     [--dbsnp dbSNP.vcf] \</div><div class="line">     [--cosmic COSMIC.vcf] \</div><div class="line">     [-L targets.interval_list] \</div><div class="line">     -o output.vcf</div></pre></td></tr></table></figure></p>
<p>在ssh shell里运行ftp,出现了一些问题，所以采用笨办法在本地windows中装了一个ftp client,下载相关文件，速度只有几十kb/s。<br>第一次选用的是hg19,文件名为ucsc.hg19.fasta，然而出现了一些问题，代码如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">java -XX:-UseCompressedClassPointers -jar /data2/TCGA/LIHC/GenomeAnalysisTK-3.6/GenomeAnalysisTK.jar \</div><div class="line">-T MuTect2 \</div><div class="line">-R /data2/TCGA/LIHC/referenceData/ucsc.hg19.fasta \</div><div class="line">-I:normal /data2/TCGA/LIHC/WXS/5c864e10-843b-41b8-ae10-2a78236b393c/TCGA-2V-A95S-10D-01D-A370-10_Illumina.bam \</div><div class="line">-I:tumor /data2/TCGA/LIHC/WXS/0a2df548-7c8c-46c5-9290-2e4d85604c17/TCGA-2V-A95S-01A-11D-A36X-10_Illumina.bam \</div><div class="line">-o /data2/TCGA/LIHC/outputVCF/output1.vcf</div></pre></td></tr></table></figure></p>
<p>备注：-XX:-UseCompressedClassPointers这个参数是因为在166.111.5.240上装java8的时候 直接java会报错……<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line">[tcga@console LIHC]$ java</div><div class="line">Error occurred during initialization of VM</div><div class="line">Could not allocate metaspace: 1073741824 bytes</div><div class="line"></div><div class="line">[tcga@console LIHC]$ java -XX:-UseCompressedClassPointers</div><div class="line">Usage: java [-options] class [args...]</div><div class="line">           (to execute a class)</div><div class="line">   or  java [-options] -jar jarfile [args...]</div><div class="line">…………</div></pre></td></tr></table></figure></p>
<p>运行结果如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div></pre></td><td class="code"><pre><div class="line">##### ERROR ------------------------------------------------------------------------------------------</div><div class="line">##### ERROR A USER ERROR has occurred (version 3.6-0-g89b7209): </div><div class="line">##### ERROR</div><div class="line">##### ERROR This means that one or more arguments or inputs in your command are incorrect.</div><div class="line">##### ERROR The error message below tells you what is the problem.</div><div class="line">##### ERROR</div><div class="line">##### ERROR If the problem is an invalid argument, please check the online documentation guide</div><div class="line">##### ERROR (or rerun your command with --help) to view allowable command-line arguments for this tool.</div><div class="line">##### ERROR</div><div class="line">##### ERROR Visit our website and forum for extensive documentation and answers to </div><div class="line">##### ERROR commonly asked questions https://www.broadinstitute.org/gatk</div><div class="line">##### ERROR</div><div class="line">##### ERROR Please do NOT post this error to the GATK forum unless you have really tried to fix it yourself.</div><div class="line">##### ERROR</div><div class="line">##### ERROR MESSAGE: Input files reads and reference have incompatible contigs. Please see https://www.broadinstitute.org/gatk/guide/article?id=63for more information. Error details: No overlapping contigs found.</div><div class="line">##### ERROR   reads contigs = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, X, Y, MT, GL000207.1, GL000226.1, GL000229.1, GL000231.1, GL000210.1, GL000239.1, GL000235.1, GL000201.1, GL000247.1, GL000245.1, GL000197.1, GL000203.1, GL000246.1, GL000249.1, GL000196.1, GL000248.1, GL000244.1, GL000238.1, GL000202.1, GL000234.1, GL000232.1, GL000206.1, GL000240.1, GL000236.1, GL000241.1, GL000243.1, GL000242.1, GL000230.1, GL000237.1, GL000233.1, GL000204.1, GL000198.1, GL000208.1, GL000191.1, GL000227.1, GL000228.1, GL000214.1, GL000221.1, GL000209.1, GL000218.1, GL000220.1, GL000213.1, GL000211.1, GL000199.1, GL000217.1, GL000216.1, GL000215.1, GL000205.1, GL000219.1, GL000224.1, GL000223.1, GL000195.1, GL000212.1, GL000222.1, GL000200.1, GL000193.1, GL000194.1, GL000225.1, GL000192.1]</div><div class="line">##### ERROR   reference contigs = [chrM, chr1, chr2, chr3, chr4, chr5, chr6, chr7, chr8, chr9, chr10, chr11, chr12, chr13, chr14, chr15, chr16, chr17, chr18, chr19, chr20, chr21, chr22, chrX, chrY, chr1_gl000191_random, chr1_gl000192_random, chr4_ctg9_hap1, chr4_gl000193_random, chr4_gl000194_random, chr6_apd_hap1, chr6_cox_hap2, chr6_dbb_hap3, chr6_mann_hap4, chr6_mcf_hap5, chr6_qbl_hap6, chr6_ssto_hap7, chr7_gl000195_random, chr8_gl000196_random, chr8_gl000197_random, chr9_gl000198_random, chr9_gl000199_random, chr9_gl000200_random, chr9_gl000201_random, chr11_gl000202_random, chr17_ctg5_hap1, chr17_gl000203_random, chr17_gl000204_random, chr17_gl000205_random, chr17_gl000206_random, chr18_gl000207_random, chr19_gl000208_random, chr19_gl000209_random, chr21_gl000210_random, chrUn_gl000211, chrUn_gl000212, chrUn_gl000213, chrUn_gl000214, chrUn_gl000215, chrUn_gl000216, chrUn_gl000217, chrUn_gl000218, chrUn_gl000219, chrUn_gl000220, chrUn_gl000221, chrUn_gl000222, chrUn_gl000223, chrUn_gl000224, chrUn_gl000225, chrUn_gl000226, chrUn_gl000227, chrUn_gl000228, chrUn_gl000229, chrUn_gl000230, chrUn_gl000231, chrUn_gl000232, chrUn_gl000233, chrUn_gl000234, chrUn_gl000235, chrUn_gl000236, chrUn_gl000237, chrUn_gl000238, chrUn_gl000239, chrUn_gl000240, chrUn_gl000241, chrUn_gl000242, chrUn_gl000243, chrUn_gl000244, chrUn_gl000245, chrUn_gl000246, chrUn_gl000247, chrUn_gl000248, chrUn_gl000249]</div><div class="line">##### ERROR ------------------------------------------------------------------------------------------</div></pre></td></tr></table></figure></p>
<p>目测是fasta文件不合适，注意到fasta文件其实需要.fasta,.fai,.dict三个文件，看了GATK提供的文件中的b37的dict的contigs貌似和bam文件里的contigs相一致，于是重新下载了b37的，这次ftp client居然飙到了100多kb/s，满足。<br>文件下载下来后运行指令:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">java -XX:-UseCompressedClassPointers -jar /data2/TCGA/LIHC/GenomeAnalysisTK-3.6/GenomeAnalysisTK.jar \</div><div class="line">-T MuTect2 \</div><div class="line">-R /data2/TCGA/LIHC/referenceData/human_g1k_v37.fasta \</div><div class="line">-I:normal /data2/TCGA/LIHC/WXS/5c864e10-843b-41b8-ae10-2a78236b393c/TCGA-2V-A95S-10D-01D-A370-10_Illumina.bam \</div><div class="line">-I:tumor /data2/TCGA/LIHC/WXS/0a2df548-7c8c-46c5-9290-2e4d85604c17/TCGA-2V-A95S-01A-11D-A36X-10_Illumina.bam \</div><div class="line">--dbsnp /data2/TCGA/LIHC/referenceData/dbsnp_132_b37.leftAligned.vcf  \</div><div class="line">--cosmic /data2/TCGA/LIHC/referenceData/b37_cosmic_v54_120711.vcf \</div><div class="line">-o /data2/TCGA/LIHC/outputVCF/output1.vcf</div></pre></td></tr></table></figure></p>
<p>一切正常，同时控制台输出实时信息Info,打印如下：<br><img src="/images/processor.png" alt="processor info"><br>重点关注一下最后两个参数，剩余的运行时间是……再见来不及挥手，本来不知道w是个什么单位，以为越来越快了，度娘一下想到是week,坐等Month[smile]<br><img src="/images/processor2.png" alt="processor info2"><br>想到–dbsnp参数和–cosmic参数是可选的，目测去掉的话会让速度快一点，试验一下<br><img src="/images/processor3.png" alt="processor info3"><br>怪我想太多，这个计算机集群好像不是很给力，top一下<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div></pre></td><td class="code"><pre><div class="line">[tcga@console outputVCF]$ top</div><div class="line">top - 11:21:58 up 56 days,  1:13,  5 users,  load average: 1.63, 2.45, 2.38</div><div class="line">Tasks: 410 total,   1 running, 408 sleeping,   1 stopped,   0 zombie</div><div class="line">Cpu(s):  0.6%us,  2.4%sy,  0.0%ni, 96.3%id,  0.5%wa,  0.0%hi,  0.1%si,  0.0%st</div><div class="line">Mem:  16431740k total, 14957408k used,  1474332k free,   155072k buffers</div><div class="line">Swap: 16386292k total,    20280k used, 16366012k free, 14252964k cached</div></pre></td></tr></table></figure></p>
<p>也就1.4G的内存，想想在自己电脑虚拟机上装java8也是不需要加-XX:-UseCompressedClassPointers 参数的，毕竟3G虚拟内存<br>询问老师后发现，这种任务是不能在管理节点上运行的，需要提交到计算节点，例如big节点上运行，将命令写入到.sh文件之中，提交的命令是qsub，流程如下：<br>创建一个.sh文件<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">touch commandTest.sh</div></pre></td></tr></table></figure></p>
<p>编辑.sh文件，写入内容<br>.sh文件中一部分是PBS作业提交所需要的代码，另一部分是进行variant analysis所需要的代码<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div></pre></td><td class="code"><pre><div class="line">#$ -cwd #</div><div class="line">#$ -o test3.log #日志文件为test3.log,保存在当前目录下</div><div class="line">#$ -l p=16      #CPU核数16</div><div class="line">#$ -l ht=big    #节点为big节点</div><div class="line">#$ -l h_vmem=2g #分配内存为2g</div><div class="line">#$ -V           #发送本地的环境变量等配置信息</div><div class="line">source ~/.bashrc</div><div class="line">export JAVA_HOME=/data2/TCGA/LIHC/java7/jdk1.7.0_79   #因为计算节点含有默认配置的JAVA,所以需要export JAVA环境变量进行覆盖，JAVA7或者JAVA8路径</div><div class="line">export PATH=$JAVA_HOME/bin:$PATH</div><div class="line">export CLASSPATH=.:$JAVA_HOME/lib/dt.jar:$JAVA_HOME/lib/tools.jar</div><div class="line">java -Xms1000m -Xmx1000m -jar /data2/TCGA/LIHC/Mutect/mutect-1.1.7.jar -T MuTect \</div><div class="line">-R /data2/TCGA/LIHC/referenceData/human_g1k_v37.fasta \        </div><div class="line">-I:normal /data2/TCGA/LIHC/WXS/5c864e10-843b-41b8-ae10-2a78236b393c/TCGA-2V-A95S-10D-01D-A370-10_Illumina.bam \</div><div class="line">-I:tumor /data2/TCGA/LIHC/WXS/0a2df548-7c8c-46c5-9290-2e4d85604c17/TCGA-2V-A95S-01A-11D-A36X-10_Illumina.bam \</div><div class="line">--dbsnp /data2/TCGA/LIHC/referenceData/dbsnp_132_b37.leftAligned.vcf \</div><div class="line">--cosmic /data2/TCGA/LIHC/referenceData/b37_cosmic_v54_120711.vcf \</div><div class="line">-L /data2/TCGA/LIHC/referenceData/Broad.human.exome.b37.interval_list \       #normal 和tummor 都是WXS的bam文件（外显子）而不是WGSM,所以需要加上 -L这个参数</div><div class="line">-o result1.txt \                                                              #输出文件为result1.txt，保存在当前目录下</div><div class="line">--coverage_file sample.coverage.wig.txt                                       #coverage信息，保存在当前目录下</div></pre></td></tr></table></figure></p>
<p>使.sh文件可执行<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">chmod +x commandTest.sh</div></pre></td></tr></table></figure></p>
<p>提交.sh文件<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">qsub commandTest.sh</div></pre></td></tr></table></figure></p>
<p>查询任务状态<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">qstat</div></pre></td></tr></table></figure></p>
<p>或者<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">qstat -j 作业号</div></pre></td></tr></table></figure></p>
<p>.log日志文件记录了代码运行的情况和剩余所需时间<br>Mutect有两个大版本，Mutect1和Mutect2，Mutect1的运行环境是JAVA7，Mutect2的运行环境是JAVA8，因此使用不同版本的Mutect时需要使用不同的JAVA环境<br>Mutect1可以检测出SNP,而Mutect2可以同时检测出SNP和indel,但是运行时间Mutect1只需要几个小时，而Mutect2需要几周的时间，我们选择Mutect1版本<br>对Mutect输出的数据进行分析：<br>txt文本文件的输出内容为：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div></pre></td><td class="code"><pre><div class="line">##MuTect:1.1.6-4-g69b7a37 Gatk:3.1-0-g72492bb</div><div class="line">contig  position    context ref_allele  alt_allele  tumor_name  normal_name score   dbsnp_site  covered power   tumor_power normal_power    normal_power_nsp    normal_power_wsp    total_reads map_Q0_reads    init_t_lod  t_lod_fstar t_lod_fstar_forward t_lod_fstar_reverse tumor_f contaminant_fraction    contaminant_lod t_q20_count t_ref_count t_alt_count t_ref_sum   t_alt_sum   t_ref_max_mapq  t_alt_max_mapq  t_ins_count t_del_count normal_best_gt  init_n_lod  normal_f    n_q20_count n_ref_count n_alt_count n_ref_sum   n_alt_sum   power_to_detect_positive_strand_artifact    power_to_detect_negative_strand_artifact    strand_bias_counts  tumor_alt_fpir_median   tumor_alt_fpir_mad  tumor_alt_rpir_median   tumor_alt_rpir_mad  observed_in_normals_count   failure_reasons judgement</div><div class="line">1   30411   TTCxTAG G   T   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   NOVEL   UNCOVERED   0   0   0   0   0   756 755 4.482505    4.077012    4.077012    0   1   0.02    -0.008773   1   0   1   0   36  0   29  0   0   GT  -352.289969 0.319149    0   256 120 8486    4082    0   0.999333    (0,0,1,0)   11  0   89  0   0   fstar_tumor_lod,possible_contamination,normal_lod,alt_allele_in_normal,poor_mapping_region_mapq0    REJECT</div><div class="line">1   69270   CTCxCTC A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   NOVEL   COVERED 0.998867    0.998867    1   1   1   1650    1487    266.034607  266.034607  266.034607  0   1   0.02    1.913288    75  0   76  0   2299    0   36  0   0   AG  -670.784921 0.29697 82  580 245 18612   8025    0   1   (0,0,76,0)  34.5    24.5    65.5    24.5    0   normal_lod,alt_allele_in_normal,poor_mapping_region_mapq0   REJECT</div><div class="line">1   69511   GTCxCAT A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   904 143 1361.541869 1361.541869 925.977581  558.118013  1   0.02    15.580048   363 0   368 0   11864   0   57  0   0   GG  -1730.950283    0.99784 383 1   462 26  15145   1   1   (1,0,265,139)   32  18  68  18  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   69897   TTCxGTG T   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   NOVEL   COVERED 0.423209    0.423478    0.999366    0.999366    0   630 592 75.71244    75.712439   0   75.712439   1   0.02    -0.184204   21  0   21  0   657 0   36  0   0   CT  -333.609287 0.376623    17  192 116 6601    3710    1   0   (0,0,0,21)  82  15  18  15  0   normal_lod,alt_allele_in_normal,poor_mapping_region_mapq0   REJECT</div><div class="line">1   877831  GCCxGGC T   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   UNCOVERED   0   0.024074    0   0.99254 0   14  0   18.915216   18.884416   7.853916    11.030501   1   0.02    -0.043862   5   0   5   0   165 0   60  0   0   CC  -33.693329  1   9   0   9   0   294 1   0.999999    (0,0,2,3)   73  21  27  21  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   881627  TCAxGGG G   A   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.997163    0.997163    1   1   1   146 0   263.897821  265.794506  165.550997  141.373198  0.985915    0.02    1.965192    70  1   70  37  2347    60  60  0   0   AA  -291.069794 1   75  0   75  0   2553    1   1   (0,1,45,36) 41  20.5    59  20.5    0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   887801  GCGxGTG A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.717212    0.717315    0.999855    0.999999    0.999855    67  0   120.584869  120.584869  78.138693   57.353529   1   0.02    -0.271952   31  0   31  0   1058    0   60  0   0   GG  -140.368469 1   36  0   36  0   1232    1   1   (0,0,20,15) 53  22  47  22  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   888639  CTCxTCC T   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   382 1   703.500535  703.500535  346.996018  481.865964  1   0.02    6.485557    178 0   175 0   6202    0   60  0   1   CC  -797.548064 1   199 0   202 0   7013    1   1   (0,0,87,120)    55  25  45  25  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   888659  CGAxCAC T   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   337 1   603.810146  603.810146  261.102853  395.680935  1   0.02    6.370155    157 0   154 0   5305    0   60  0   0   CC  -685.931158 1   176 0   177 0   6016    1   1   (0,0,65,103)    63  17.5    37  17.5    0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   897325  TGCxCAC G   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.999622    0.999622    1   1   1   175 0   320.405654  320.405654  195.845077  163.429944  1   0.02    1.842381    84  0   82  0   2813    0   60  0   0   CC  -350.219495 1   90  0   90  0   3073    1   1   (0,0,50,42) 58  19.5    42  19.5    0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   908325  CGCxCCA A   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   NOVEL   COVERED 0.999907    0.999907    1   1   1   185 0   -11.87038   6.915099    0   23.074697   0.055556    0.02    1.565211    87  85  5   2662    128 60  60  0   0   AC  -13.169275  0.21978 70  71  20  2091    316 0.629966    0.95308 (86,15,0,12)    11  10  89  10  0   possible_contamination,normal_lod,alt_allele_in_normal,strand_artifact  REJECT</div><div class="line">1   909238  CACxTGG G   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.976865    0.976865    1   1   1   111 0   217.326209  217.326209  150.49395   82.440037   1   0.02    2.205302    54  0   55  0   1911    0   60  0   0   CC  -220.807568 1   56  0   56  0   1941    1   1   (0,0,39,20) 41  23  59  23  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   949654  GGTxCGG A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   406 0   722.758598  722.758598  277.459966  494.59709   1   0.02    6.467777    178 0   182 0   6361    0   60  0   0   GG  -888.518804 1   223 0   224 0   7817    1   1   (0,0,71,124)    59  24  41  24  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   981931  ATCxCGA A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.999622    0.999622    1   1   1   159 0   329.54859   329.54859   72.803779   268.375543  1   0.02    1.942183    84  0   84  0   2895    0   60  0   0   GG  -295.794547 1   74  0   74  0   2605    1   1   (0,0,21,66) 49  26.5    51  26.5    0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   982994  TTTxCGG T   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   487 0   852.878057  852.878057  428.032505  523.242378  1   0.02    8.422372    218 0   217 0   7494    0   60  0   0   CC  -1034.224837    1   262 0   267 0   9071    1   1   (0,0,107,136)   49  23  51  23  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   984302  CACxCTC T   C   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.999993    0.999993    1   1   1   191 0   423.824351  423.824351  285.86096   182.179804  1   0.02    4.211996    108 0   110 0   3715    0   60  0   0   CC  -314.305426 1   80  0   81  0   2758    1   1   (0,0,74,48) 56  21  44  21  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   990280  GGAxGTG C   T   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   290 0   467.279822  467.279822  307.301639  206.474519  1   0.02    3.447023    127 0   127 0   4068    0   60  0   0   TT  -567.135033 0.987578    157 2   159 74  4998    1   1   (0,0,84,56) 53  20  47  20  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1147422 GGTxTCC C   T   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.999622    0.999622    1   1   1   177 0   142.727422  142.985799  92.848027   60.573629   0.559524    0.02    1.443973    85  37  47  1305    1456    60  60  0   0   CT  -163.714033 0.597826    90  37  55  1271    1652    1   1   (21,24,31,20)   36  23  64  23  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1158631 CTCxTCC A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.997163    0.997163    1   1   1   136 0   261.344811  261.344811  204.042048  94.171217   1   0.02    1.96524 71  0   71  0   2275    0   60  0   0   GG  -236.948829 1   63  0   64  0   2066    1   1   (0,0,58,23) 34  19  66  19  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1178245 TAGxGTG C   T   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.999835    0.999835    1   1   1   202 0   134.576074  134.586206  94.682931   54.57785    0.488889    0.02    1.49976 89  46  44  1530    1407    60  60  0   0   CT  -175.038722 0.513514    109 54  57  1837    1813    1   1   (27,25,31,18)   40  20  60  20  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1178482 CACxCAC A   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.999995    0.999995    1   1   1   227 0   122.274729  124.285428  47.151148   89.09402    0.357143    0.02    4.284572    112 72  40  2269    1369    60  60  0   0   AG  -181.298434 0.491228    113 58  56  1833    1889    1   0.999998    (19,61,16,28)   52  25  48  25  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1178925 TGGxCAG G   A   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 1   1   1   1   1   339 0   227.062104  227.175947  88.612151   158.816115  0.470968    0.02    5.498553    156 82  73  2932    2389    60  60  0   0   AG  -288.490333 0.513661    182 89  94  3118    2987    1   1   (30,63,28,52)   63  24  37  24  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1226757 GGCxGCC G   A   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.986153    0.986153    1   1   1   125 0   86.467527   86.502219   57.870681   35.770359   0.474576    0.02    1.87139 58  31  28  1012    909 60  60  0   0   AG  -104.933334 0.515152    64  32  34  1025    1086    1   1   (12,20,17,13)   48.5    23  51.5    23  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div><div class="line">1   1254841 GGGxCCG C   G   TCGA-2V-A95S-01A-11D-A36X-10    TCGA-2V-A95S-10D-01D-A370-10    0   DBSNP   COVERED 0.739671    0.739671    1   1   1   82  0   38.659769   38.794746   3.347288    35.293017   0.433333    0.02    -0.279264   30  17  13  589 416 60  60  0   0   CG  -85.951008  0.52    49  24  26  802 887 0.999993    0.956562    (4,14,2,11) 70  16  30  16  0   germline_risk,normal_lod,alt_allele_in_normal   REJECT</div></pre></td></tr></table></figure></p>
<p>GATK presentation文档给出了输出内容中比较重要的一些信息：<br><img src="/image/mutectOutput.png" alt="mutectOutput"><br>Context:一些癌症类型的癌变特征<br>t_lod_fstar:最后调整过的癌变LOD值<br>tumor_f:tumor中的等位基因变体分数<br>strand_bias_counts:<br>observed_in_normals_count:<br>failure_reasons:<br>judgement:</p>

      
    </div>
    
  </div>
  
    
    <div class="copyright">
        <p><span>本文标题:</span><a href="/2016/11/29/GATK-bamfile/">GATK_bamfile</a></p>
        <p><span>文章作者:</span><a href="/" title="回到主页">紫色的紫</a></p>
        <p><span>发布时间:</span>2016-11-29, 15:52:15</p>
        <p><span>最后更新:</span>2016-12-18, 18:51:18</p>
        <p>
            <span>原始链接:</span><a class="post-url" href="/2016/11/29/GATK-bamfile/" title="GATK_bamfile">http://yoursite.com/2016/11/29/GATK-bamfile/</a>
            <span class="copy-path" data-clipboard-text="原文: http://yoursite.com/2016/11/29/GATK-bamfile/　　作者: 紫色的紫" title="点击复制文章链接"><i class="fa fa-clipboard"></i></span>
            <script> var clipboard = new Clipboard('.copy-path'); </script>
        </p>
        <p>
            <span>许可协议:</span><i class="fa fa-creative-commons"></i> <a rel="license" href="http://creativecommons.org/licenses/by-nc-sa/4.0/" title="CC BY-NC-SA 4.0 International" target = "_blank">"署名-非商用-相同方式共享 4.0"</a> 转载请保留原文链接及作者。
        </p>
    </div>



    <nav id="article-nav">
        
            <div id="article-nav-newer" class="article-nav-title">
                <a href="/2016/11/29/HaplotypeCaller/">
                    HaplotypeCaller
                </a>
            </div>
        
        
            <div id="article-nav-older" class="article-nav-title">
                <a href="/2016/11/22/GATK-Schedule/">
                    GATK_Schedule
                </a>
            </div>
        
    </nav>

  
</article>

    <div id="toc" class="toc-article">
        <strong class="toc-title">文章目录</strong>
        
            
        
    </div>
    <style>
        .left-col .switch-btn,
        .left-col .switch-area {
            display: none;
        }
        .toc-level-3 i,
        .toc-level-3 ol {
            display: none !important;
        }
    </style>

    <input type="button" id="tocButton" value="隐藏目录"  title="点击按钮隐藏或者显示文章目录">

    <script>
        yiliaConfig.toc = ["隐藏目录", "显示目录", !!"false"];
    </script>



    
<div class="share">
    
        <div class="bdsharebuttonbox">
            <a href="#" class="fa fa-twitter bds_twi" data-cmd="twi" title="分享到推特"></a>
            <a href="#" class="fa fa-weibo bds_tsina" data-cmd="tsina" title="分享到新浪微博"></a>
            <a href="#" class="fa fa-qq bds_sqq" data-cmd="sqq" title="分享给 QQ 好友"></a>
            <a href="#" class="fa fa-files-o bds_copy" data-cmd="copy" title="复制网址"></a>
            <a href="#" class="fa fa fa-envelope-o bds_mail" data-cmd="mail" title="通过邮件分享"></a>
            <a href="#" class="fa fa-weixin bds_weixin" data-cmd="weixin" title="生成文章二维码"></a>
            <a href="#" class="fa fa-share-alt bds_more" data-cmd="more"></i></a>
        </div>
        <script>
            window._bd_share_config={
                "common":{"bdSnsKey":{},"bdText":"GATK_bamfile　| Purple Violet　","bdMini":"2","bdMiniList":false,"bdPic":"","bdStyle":"0","bdSize":"24"},"share":{}};with(document)0[(getElementsByTagName('head')[0]||body).appendChild(createElement('script')).src='http://bdimg.share.baidu.com/static/api/js/share.js?v=89860593.js?cdnversion='+~(-new Date()/36e5)];
        </script>
    

    
</div>







    




    <div class="scroll" id="post-nav-button">
        
            <a href="/2016/11/29/HaplotypeCaller/" title="上一篇: HaplotypeCaller">
                <i class="fa fa-angle-left"></i>
            </a>
        

        <a title="文章列表"><i class="fa fa-bars"></i><i class="fa fa-times"></i></a>

        
            <a href="/2016/11/22/GATK-Schedule/" title="下一篇: GATK_Schedule">
                <i class="fa fa-angle-right"></i>
            </a>
        
    </div>

    <ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2016/12/01/query-py/">query_py</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/29/HaplotypeCaller/">HaplotypeCaller</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/29/GATK-bamfile/">GATK_bamfile</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/22/GATK-Schedule/">GATK_Schedule</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/05/ajax_study/">ajax_study</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/03/phphp_study/">phphp_study</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/03/Search_php3/">Search.php（3）</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/11/01/Search_php2/">Search.php（2）</a></li><li class="post-list-item"><a class="post-list-link" href="/2016/10/31/Search_php1/">Search.php（1）</a></li></ul>




    <script>
        
    </script>
</div>
      <footer id="footer">
    <div class="outer">
        <div id="footer-info">
            <div class="footer-left">
                <i class="fa fa-copyright"></i> 
                2016 紫色的紫
            </div>
            <div class="footer-right">
                <a href="http://hexo.io/" target="_blank" title="快速、简洁且高效的博客框架">Hexo</a>  Theme <a href="https://github.com/MOxFIVE/hexo-theme-yelee" target="_blank" title="简而不减 Hexo 双栏博客主题  v3.5">Yelee</a> by MOxFIVE <i class="fa fa-heart animated infinite pulse"></i>
            </div>
        </div>
        
            <div class="visit">
                
                    <span id="busuanzi_container_site_pv" style='display:none'>
                        <span id="site-visit" title="本站到访数"><i class="fa fa-user" aria-hidden="true"></i><span id="busuanzi_value_site_uv"></span>
                        </span>
                    </span>
                
                
                    <span>| </span>
                
                
                    <span id="busuanzi_container_page_pv" style='display:none'>
                        <span id="page-visit"  title="本页阅读量"><i class="fa fa-eye animated infinite pulse" aria-hidden="true"></i><span id="busuanzi_value_page_pv"></span>
                        </span>
                    </span>
                
            </div>
        
    </div>
</footer>
    </div>
    
<script data-main="/js/main.js" src="//cdn.bootcss.com/require.js/2.2.0/require.min.js"></script>

    <script>
        $(document).ready(function() {
            var iPad = window.navigator.userAgent.indexOf('iPad');
            if (iPad > -1 || $(".left-col").css("display") === "none") {
                var bgColorList = ["#9db3f4", "#414141", "#e5a859", "#f5dfc6", "#c084a0", "#847e72", "#cd8390", "#996731"];
                var bgColor = Math.ceil(Math.random() * (bgColorList.length - 1));
                $("body").css({"background-color": bgColorList[bgColor], "background-size": "cover"});
            }
            else {
                var backgroundnum = 1;
                var backgroundimg = "url(/background/bg-x.jpg)".replace(/x/gi, Math.ceil(Math.random() * backgroundnum));
                $("body").css({"background": backgroundimg, "background-attachment": "fixed", "background-size": "cover"});
            }
        })
    </script>





<div class="scroll" id="scroll">
    <a href="#" title="返回顶部"><i class="fa fa-arrow-up"></i></a>
    <a href="#comments" onclick="load$hide();" title="查看评论"><i class="fa fa-comments-o"></i></a>
    <a href="#footer" title="转到底部"><i class="fa fa-arrow-down"></i></a>
</div>
<script>
    // Open in New Window
    
        var oOpenInNew = {
            
            
            
            
            
            
             archives: ".archive-article-title", 
             miniArchives: "a.post-list-link", 
            
             friends: "#js-friends a", 
             socail: ".social a" 
        }
        for (var x in oOpenInNew) {
            $(oOpenInNew[x]).attr("target", "_blank");
        }
    
</script>

<script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js">
</script>
  </div>
</body>
</html>